crates/sallyport/src/guest/handler.rs

```diff
@@ -254,6 +267,54 @@ pub trait Handler {
         self.execute(syscall::Fstat { fd, statbuf })?
     }
 
+    /// Executes [`futex`](https://man7.org/linux/man-pages/man2/futex.2.html) syscall.
+    fn futex(
+        &mut self,
+        uaddr: &mut AtomicU32,
+        futex_op: c_int,
+        val: u32,
+        timespec: Option<&timespec>,
+        _uaddr2: Option<&mut AtomicU32>,
+        val3: u32,
+    ) -> Result<c_long> {
+        let futex_op = futex_op & !FUTEX_PRIVATE_FLAG;
```

Comment:
Why silently drop the flag and not return `ENOTSUP` instead?
At least there should be a comment explaining why this handled this way.